src/lib/components/Callback/Callback.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Actions, SupAuth, history } from 'gisida';
+
+class Callback extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { loaded: false };
+    this.history = history;
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (!this.state.loaded
+      && nextProps.global.APP
+      && nextProps.global.APP.loaded) {
+      this.setState({ 
+        loaded: true,
+        APP: { ...nextProps.global.APP },
+       }, () => {
+        this.authorizeUser(this.state.APP);
+      });
+    }
+  }
+
+  async authorizeUser(APP) {
+    const { dispatch } = this.props;
+    const accessToken = this.getAccessToken();
+    dispatch(Actions.receiveToken(accessToken));
+
+    const userIsAuthorized = await SupAuth.authorizeUser(APP, accessToken);
+
+    if (!userIsAuthorized || !accessToken) {
+      localStorage.removeItem('access_token');
+      localStorage.removeItem('user');
+      localStorage.removeItem('authConfig');
```

Comment:
@KipSigei - At this point in the code, those items should already be removed from localStorage by [Gisida's SupAuth utility](https://github.com/onaio/gisida/blob/signin-auth-dev/src/utils/supAuth.js#L91)'s `defaultUnSupAuthZ` function.

In general, I tried keeping everything related to the Supplemental Authorization/Authentication flow in the util file, while everything ONA Oauth2 is in the connector modules, so the only thing in Gisida-React which deals with anything Auth is this Callback component and the [Private Route](https://github.com/onaio/gisida-react/blob/a161a0c13a0fa9566f19da4deb217f9a1fe1fe2a/src/lib/routes/privateRoute.js#L7) (which the [User passes in from index.js](https://github.com/onaio/service-mapping/blob/signin-auth-dev/src/index.js#L86), in this case the SupAuth utility's `defaultSupViewAuthC` function).